ssas/SwaggerStructs.go

```diff
@@ -0,0 +1,222 @@
+
+import (
+	"time"
+)
+
+// The requested operation was successful.  There is no body returned.
+// swagger:response okResponse
+type OkResponse struct {}
+
+// The request is invalid or malformed
+// swagger:response badRequestResponse
+type BadRequestResponse struct {
+	// Invalid request
+	// in: body
+	Body ErrorResponse
+}
+
+// Required credentials are missing or invalid
+// swagger:response invalidCredentials
+type InvalidCredentialsResponse struct {
+	// Invalid credentials
+	// in: body
+	Body ErrorResponse
+}
+
+// An internal server error has occurred
+// swagger:response serverError
+type ServerErrorResponse struct {
+	// Internal server error
+	// in: body
+	Body ErrorResponse
+}
+
+// The specified resource is not found
+// swagger:response notFoundResponse
+type NotFoundResponse struct {
+	// Resource not found
+	// in: body
+	Body ErrorResponse
+}
+
+type ErrorResponse struct {
+	// Error type
+	// Required: true
+	Error				string
+	// More information about the error
+	ErrorDescription	string
+}
+
+// The successfully created/altered system is returned
+// swagger:response systemResponse
+type SystemResponse struct {
+	// System details
+	// in: body
+	Body struct {
+		// The client ID for this system
+		// Required: true
+		ClientID		string
+		// The client secret for this system
+		// Required: true
+		ClientSecret	string
+		// This system's ID
+		// Required: true
+		SystemID		string
+		// The user-specified name for the system
+		// Required: true
+		ClientName		string
+		// The expiration date for these credentials
+		// Required: true
+		ExpiresAt		time.Time
+		// Optional IP addresses from which this system is allowed to connect
+		// Required: false
+		IPs				[]string	`json:"ips,omitempty"`
+	}
+}
+
+// The group was successfully created/altered
+// swagger:response groupResponse
+type GroupResponse struct {
+	// Group details
+	// in: body
+	Body struct {
+		// The group's ID
+		// Required: true
+		ID        int
+		// Creation timestamp for the group
+		// Required: false
+		CreatedAt time.Time
+		// Last update timestamp for the group
+		// Required: false
+		UpdatedAt time.Time
+		// The date at which the group was deleted.  This is unlikely to be present in API output.
+		// Required: false
+		DeletedAt time.Time
+		// The user-provided identifier for the group
+		// Required: true
+		GroupID   string
+		// The user-provided data for the group, which should be associated with all systems in this group.
+		// Required: true
+		XData     string `json:"xdata"`
+		// A parsed version of the user-provided data
+		// Required: true
+		Data      GroupSummary
+	}
+}
+
+// The specified system has a public key, which is returned
+// swagger:response publicKeyResponse
+type PublicKeyResponse struct {
+	// Public key details
+	// in: body
+	Body struct {
+		// This system's client ID
+		// Required: true
+		ClientID	string	`json:"client_id"`
+		// The public key (if any) registered for this system
+		// Required: true
```

Comment:
can this be required if the public key is not guaranteed to be there?

Reply:
Yes!  If there's no public key, the response looks like:
  `{"client_id":"this_client_id","public_key":""}`